### PR TITLE
Refactor table's `persist` to persist asynchronously before advancing…

### PR DESCRIPTION
**Pull request by eyalsol**
… fixed windows

* ML-1019: Persist by key before advancing the window
* ML-1055: Refactor Query (service.get). fixed windows should return last closed window

Signed-off-by: Eyal Salomon <eyals@iguazio.com>

### CODE REVIEW

File: storey/table.py [hunk 1]
```diff
@@ -1540,6 +1585,12 @@ def __init__(self):
 class _PersistJob:
     def __init__(self, key, data, callback, extra_data=None):
         self.key = key
-        self.data = data
+        self.data = copy.deepcopy(data)
```
Comment (gtopper) [inline storey/table.py:1588]: Looks like this is used only from NoSqlTarget, when periodic flushing is disabled. So why do we need to deepcopy it?

File: storey/table.py [hunk 2]
```diff
@@ -254,6 +253,15 @@ def _aggregates_to_schema(self):
 
         return schema
 
+    def _flush_pending(self, key):
+        aggregations_attrs = self._get_aggregations_attrs(key)
+        if not aggregations_attrs:
+            return
+        for _, bucket in aggregations_attrs.aggregation_buckets.items():
```
Comment (gtopper) [inline storey/table.py]: `items()` -> `values()`

**Review by gtopper — COMMENTED** [review]
Any performance measurements? This could have a significant impact, especially when the window is advanced often relative to the sample time interval.

File: storey/table.py (hunk 1 above)
Comment (eyalsol) [reply storey/table.py:1588]: I don't want a mixture of copied/not copied members. so I deep copy all

**Review by dinal — APPROVED** [review]

**Pull request merged**